cogs/voice_channels.py

```diff
@@ -76,6 +80,90 @@ async def leave_help(self, ctx):
+        docstring = """
+                ```Allows the user to specify a time, after which they will be automatically removed from their current VC.
+                Admins can mention a user to set a timer for them.
+
+                Arguments:
+                    time: The amount of time to wait before removing a user, in minutes. Defaults to 30 minutes.
+                Examples:
+                    c.sleep 
+                    c.sleep time=45 @pip```
+                """
+        docstring = self.bot.remove_indentation(docstring)
+        await ctx.send(docstring)
+
+    @commands.command(name=f"sleep")
+    async def sleep_timer(self, ctx):
+        if not await self.bot.has_perm(ctx): return False
+        """
+        Allows the user to specify a time, after which they will be automatically removed from their current VC.
+        Admins can mention a user to set a timer for them.
+
+        Arguments:
+            time: The amount of time to wait before removing a user, in minutes. Defaults to 30 minutes.
+        Examples:
+            c.sleep
+            c.sleep time=45
+        """
+        message = ctx.message
+        content = message.content
+        time = float(self.bot.get_variable(content, "time", type="float", default=30))
+        user = self.bot.admin_override(ctx)
+
+        # Checks if user is in a VC.
+        voice = user.voice
+        if voice:
+            channel = voice.channel
+        else:
+            if user == ctx.author:
+                await ctx.send("Join a voice channel before using this command.")
+                return
+            else:
+                await ctx.send("The mentioned user must be in a voice channel.")
+                return
+
+        # Check if there's already an entry in the database for this user
+        self.bot.cursor.execute(
+            f"SELECT rowid, * FROM sleep_timer WHERE"
+            f" user_id={user.id} AND server={ctx.guild.id}")
+        result = self.bot.cursor.fetchone()
+
+        if time > 0:
+            time = (max(min(1440, time), 0.016)) / 60  # Limit to 1 day or 1 second. Also converts from the given minutes to hours.
+            end_time = self.bot.hours_from_now(time)
+            time_string = self.bot.time_to_string(hours=time)
+
+            if result:  # Update existing entry.
+                self.bot.cursor.execute(f"UPDATE sleep_timer SET end_time={end_time} WHERE rowid={result[0]}")
+            else:  # Create entry.
+                self.bot.cursor.execute(
+                    f"INSERT INTO sleep_timer VALUES({ctx.guild.id}, {user.id}, {end_time})")
+            self.bot.cursor.execute("commit")
+
+            await ctx.send(f"{user.name} will automatically be removed from VC in {time_string}.")
+        else:
+            await ctx.send("Provided time must be above 0.")
+            return
+
+        try:
+            self.bot.cursor.execute("commit")
+        except:
+            pass
+
+    def init_db(self, cursor):
```

Comment:
Although you've written a function for init_db, you never actually called it in this object's \_\_init\_\_ function, meaning it is never run.